web/src/components/workflow-editor/workflow-editor.tsx

```diff
@@ -96,89 +110,111 @@ export default function WorkflowEditor({
 		router,
 	]);
 
+	useEffect(() => {
+		if (!workflow || !encryptedSecrets) {
+			return;
+		}
+		workflow.nodes.forEach((node, nodeIdx) => {
+			console.log("node", node);
+			if (node.type === "action") {
+				const secretName = Object.values(node.secretsMapping)[0];
+				if (
+					secretName &&
+					!encryptedSecrets.find((s) => s.secretId === secretName)
```

Comment:
Minor: imo it is a bit cleaner to build a set of the secret ids before iterating over the nodes because then you can perform a lookup in constant time instead of always iterating here over all the encrypted secrets.